File: service-security/api/src/main/java/org/delivery/api/domain/token/ifs/TokenHelperIfs.java
package org.delivery.api.domain.token.ifs;

import org.delivery.api.domain.token.model.TokenDto;

import java.util.Map;

public interface TokenHelperIfs {

    TokenDto issueAccessToken(Map<String, Object> claims);

    TokenDto issueRefreshToken(Map<String, Object> claims);

    Map<String, Object> validationTokenWithThrow(String token);
}

File: service-security-kotlin-msa/account/src/main/kotlin/org/delivery/account/domain/token/ifs/TokenHelperIfs.kt
package org.delivery.account.domain.token.ifs

import org.delivery.account.domain.token.model.TokenDto

interface TokenHelperIfs {
    fun issueAccessToken(claims: Map<String, Any>): TokenDto
    fun issueRefreshToken(claims: Map<String, Any>): TokenDto
    fun validationTokenWithThrow(token: String?): Map<String, Any>
}